### ISSUE TITLE
Feature request: rename and reposition card types

### ISSUE THREAD

**Issue by sudomain** [ankidroid/Anki-Android#6017]
I couldn't find existing issues for these things so here it goes:

A couple desktop features I'd like to port/submit a PR for are related to the card type editor:
1.Rename Card Type
2. Reposition Card Type

Add and remove card types are already supported. For completeness: 
1. Deck override setting is issue #5396 
2. Browser appearance overriding for the front and back templates is issue #5158

**Comment by mikehardy** [comment]
Interesting. Yes now they are just CardN where N is some number, and you can't move them. This would be an interesting feature but it must wait for #5151 to land as that changes everything in the area. I'm hoping for that in the next day or three but it's open source so I can't promise sorry

**Comment by sudomain** [comment]
All good. Thanks for all of your contributions

**Comment by sudomain** [comment]
Linking to a couple areas of interest pertaining to this which I'll use once #5151 goes through.

[method for getting the current template name](https://github.com/ankidroid/Anki-Android/blob/31b8a2a798bf1dfb2d13ce4dcff10ae836137748/AnkiDroid/src/main/java/com/ichi2/anki/CardTemplateEditor.java#L334)

[cloze models have one card type, thus no need for reposition. Check if desktop allows renaming of the card type](https://github.com/ankidroid/Anki-Android/blob/31b8a2a798bf1dfb2d13ce4dcff10ae836137748/AnkiDroid/src/main/java/com/ichi2/anki/CardTemplateEditor.java#L437)

[The delete card checks if there's 1 card type left. We could probably extract/reuse this code for reposition](https://github.com/ankidroid/Anki-Android/blob/31b8a2a798bf1dfb2d13ce4dcff10ae836137748/AnkiDroid/src/main/java/com/ichi2/anki/CardTemplateEditor.java#L463)

In fact, a lot of the add/delete card type functionality might be reuseable (add checks for names of existing card types, delete must handle changing the `ord` variable for each card type that comes after the card type currently being deleted)

**Comment by sudomain** [comment]
Since #5151 was merged may I make an attempt at these features?

**Comment by mikehardy** [comment]
Yep - 5151 was definitely blocking, and this is definitely unblocked now. CardTemplateEditor is open for business for anyone else that wants to do it

**Comment by sudomain** [comment]
Hi, I don't have any new information about these feature requests (other than it's much harder than I thought to implement reposition and keep the ability to 'Undo' changes via TemporaryModel.java). I'd like to politely request this be reopened and the GH `Keep open` label be added since these are desktop features.

**Comment by mikehardy** [comment]
Sure, ecosystem compatibility is always a goal. This area of the code (CardTemplateEditor) is ridiculously hard, yes

**Comment by sudomain** [comment]
Thank you for adding the Keep Open label, but from my device it's still showing the issue as closed :)

**Comment by mikehardy** [comment]
Bah, oversight on my part :-). Reopened

**Comment by MaroonShaded** [comment]
This should be implemented. I can't rename my card types and that makes it a bit unorganized. Although there is a workaround for this, my card types were created while the app was in different languages and they don't match.
I don't know if it's a good idea to comment on this 2 year old issue.

**Comment by david-allison** [comment]
Cloze and image occlusion notes cannot be renamed
